File: 2017/5/main.py
import fileinput

jumps = []
jumps2 = []
for line in fileinput.input():
    jumps.append(int(line))
    jumps2.append(int(line))

i = 0
s1 = 0
while i < len(jumps) and i >= 0:
    a = jumps[i]
    jumps[i] += 1
    i += a
    s1 += 1

print(s1)

i = 0
s2 = 0
while i < len(jumps2) and i >= 0:
    a = jumps2[i]
    if a >= 3:
        jumps2[i] -= 1
    else:
        jumps2[i] += 1
    i += a
    s2 += 1
print(s2)

File: 2017/5/main.c
#include<stdio.h>
#include<string.h>
#include <stdlib.h>

int main() {
  int jumps[2048];
  int jumps2[2048];
  int s1, s2 = 0;
  int size = 0;

  while (1) {
    char line[64];
    fgets(line, 64, stdin);
    if (line == '\0') break;

    jumps[size] = atoi(line);
    jumps2[size] = atoi(line);
    size++;
    printf("%d %d\n", size, jumps[size]);
  }
  size++;

  // printf("%d %d %d %d \n", jumps[0], jumps[1], jumps[2], jumps[3]);

  // part 1
  // int i = 0;
  // while(i >= 0 && i < size) {
  //   int a = jumps[i];
  //   jumps[i]++;
  //   s1++;
  //   i += a;
  // }


  printf("%d\n", s1);
  printf("%d\n", s2);
}
